src/BenchmarkGrainInterfaces/MapReduce/DataflowGrainsInterfaces.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans;
+
+namespace GrainInterfaces
+{
+    public interface IDataflowGrain : IGrainWithGuidKey
```

Comment:
Almost all of the code is needed for this benchmark in one form or another. But writing it in such way leaves the ability for it to evolve into something bigger than just a sample (i wouldn't add epithet 'cool' yet).